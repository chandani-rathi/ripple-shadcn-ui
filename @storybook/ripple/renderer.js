// renderer.js
export async function renderToCanvas({ storyFn, canvasElement }) {
  console.log("renderer")
  const output = storyFn();

  // Example: simple HTML render
  if (typeof output === 'string') {
    canvasElement.innerHTML = output;
  } else if (output instanceof HTMLElement) {
    canvasElement.innerHTML = '';
    canvasElement.appendChild(output);
  } else {
    canvasElement.innerHTML = `<pre>Unsupported output: ${JSON.stringify(output)}</pre>`;
  }
}
