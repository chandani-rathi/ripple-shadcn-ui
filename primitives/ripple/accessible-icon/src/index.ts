export * from './accessible-icon.ripple';