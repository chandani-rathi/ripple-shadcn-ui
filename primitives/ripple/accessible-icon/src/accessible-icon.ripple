import { trackSplit } from "ripple"
import { VisuallyHidden } from "@ripple-primitives/visually-hidden"
const NAME = 'ACCESSIBLE_ICON';



export component AccessibleIcon(props){
  const accessibilityProps = {
    'aria-hidden': 'true',
    focusable: 'false', // See: https://allyjs.io/tutorials/focusing-in-svg.html#making-svg-elements-focusable
  }
  const [children, label, rest] = trackSplit(props, ['children', 'label'])
  <@children {...@rest} {...accessibilityProps} />
  <VisuallyHidden>{@label}</VisuallyHidden>
}