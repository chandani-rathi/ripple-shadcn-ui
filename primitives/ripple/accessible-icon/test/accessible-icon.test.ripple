import { axe } from 'vitest-axe';
import userEvent from '@testing-library/user-event';
import { AccessibleIcon }  from "../src/accessible-icon.ripple";

const TEXT_CHILD = "TEST";

component CrossIcon(props) {
     <svg viewBox="0 0 32 32" width={24} height={24} fill="none" stroke="currentColor" {...props}>
        <path d="M2 30 L30 2 M30 30 L2 2" />
    </svg>
}

describe("accessible-icon", () => {
    beforeEach(() => {
		component Test(){
			<AccessibleIcon label={TEXT_CHILD} data-id={TEXT_CHILD}>
				component children(props){
					<CrossIcon {...props} />
				}
			</AccessibleIcon>
		}
		render(Test);
	});

    it('should have no accessibility violations', async () => {
		expect(await axe(container)).toHaveNoViolations();
	});
    
    it("should work", () => {
         expect(container).toMatchSnapshot()
    })
})