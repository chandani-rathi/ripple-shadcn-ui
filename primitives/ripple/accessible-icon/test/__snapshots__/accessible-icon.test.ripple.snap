// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`accessible-icon > should work 1`] = `
<div>
  <!---->
  <!---->
  <!---->
  <svg
    aria-hidden="true"
    data-id="TEST"
    fill="none"
    focusable="false"
    height="24"
    stroke="currentColor"
    viewBox="0 0 32 32"
    width="24"
  >
    <path
      d="M2 30 L30 2 M30 30 L2 2"
    />
  </svg>
  <!---->
  <!---->
  <!---->
  <!---->
  <span
    style="position: absolute; border: 0px; padding: 0px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
  >
    <!---->
    <!---->
    TEST
    <!---->
    <!---->
    <!---->
  </span>
  <!---->
  <!---->
  <!---->
  <!---->
  
</div>
`;
